src/arch/x86/analyzex86.rs

```diff
@@ -331,65 +333,32 @@ impl Statex86 {
             return 0;
         }
         let index: isize = index_check;
```

Comment:
the line that produces `index_check` is fairly inscrutable to me. I initially read it as testing that `address + address_width + stack_address` wouldn't overflow, but in refactoring got something wrong which broke all stack reads.. So I left it as-is, but if it's checking for overflow I think it would be much easier to read as some appropriate incantation of `.checked_sub` (like in `read_int`)